frame/assets/src/lib.rs

```diff
@@ -147,80 +136,82 @@ pub trait Trait: frame_system::Trait {
-		/// # </weight>
-		#[weight = 0]
-		fn issue(origin, #[compact] total: T::Balance) {
-			let origin = ensure_signed(origin)?;
+	/// The origin which may forcibly create or destroy an asset.
+	type ForceOrigin: EnsureOrigin<Self::Origin>;
 
-			let id = Self::next_asset_id();
-			<NextAssetId<T>>::mutate(|id| *id += One::one());
+	/// The basic amount of funds that must be reserved when creating a new asset class.
+	type AssetDepositBase: Get<BalanceOf<Self>>;
 
-			<Balances<T>>::insert((id, &origin), total);
-			<TotalSupply<T>>::insert(id, total);
+	/// The additional funds that must be reserved for every zombie account that an asset class
+	/// supports.
+	type AssetDepositPerZombie: Get<BalanceOf<Self>>;
 
-			Self::deposit_event(RawEvent::Issued(id, origin, total));
-		}
+	/// Weight information for extrinsics in this pallet.
+	type WeightInfo: WeightInfo;
+}
 
-		/// Move some assets from one holder to another.
-		///
-		/// # <weight>
-		/// - `O(1)`
-		/// - 1 static lookup
-		/// - 2 storage mutations (codec `O(1)`).
-		/// - 1 event.
-		/// # </weight>
-		#[weight = 0]
-		fn transfer(origin,
-			#[compact] id: T::AssetId,
-			target: <T::Lookup as StaticLookup>::Source,
-			#[compact] amount: T::Balance
-		) {
-			let origin = ensure_signed(origin)?;
-			let origin_account = (id, origin.clone());
-			let origin_balance = <Balances<T>>::get(&origin_account);
-			let target = T::Lookup::lookup(target)?;
-			ensure!(!amount.is_zero(), Error::<T>::AmountZero);
-			ensure!(origin_balance >= amount, Error::<T>::BalanceLow);
+#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug)]
+pub struct AssetDetails<
+	Balance: Encode + Decode + Clone + Debug + Eq + PartialEq,
+	AccountId: Encode + Decode + Clone + Debug + Eq + PartialEq,
+	DepositBalance: Encode + Decode + Clone + Debug + Eq + PartialEq,
+> {
+	/// Can change `owner`, `issuer`, `freezer` and `admin` accounts.
+	owner: AccountId,
+	/// Can mint tokens.
+	issuer: AccountId,
+	/// Can thaw tokens, force transfers and burn tokens from any account.
+	admin: AccountId,
+	/// Can freeze tokens.
+	freezer: AccountId,
+	/// The total supply across all accounts.
+	supply: Balance,
+	/// The balance deposited for this asset.
+	///
+	/// This pays for the data stored here together with any virtual accounts.
+	deposit: DepositBalance,
+	/// The number of balance-holding accounts that this asset may have, excluding those that were
+	/// created when they had a system-level ED.
+	max_zombies: u32,
+	/// The ED for virtual accounts.
+	min_balance: Balance,
+	/// The current number of zombie accounts.
+	zombies: u32,
+	/// The total number of accounts.
+	accounts: u32,
+}
 
-			Self::deposit_event(RawEvent::Transferred(id, origin, target.clone(), amount));
-			<Balances<T>>::insert(origin_account, origin_balance - amount);
-			<Balances<T>>::mutate((id, target), |balance| *balance += amount);
-		}
+#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default)]
+pub struct AccountData<
```

Comment:
This is gonna screw with any frontend types (like in polkadot-js) if both this and `balances::AccountData` are present in the runtime. @JesseAbram and I ran into this when trying to deploy this pallet in a node-template.
Let's rename it so people are free to include both pallets.